### PR TITLE
[GEARPUMP-285] fix false alarm of shutting down executor time out

### PR DESCRIPTION
**Pull request by huafengw**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [ ] Make sure tests pass via `sbt clean test`.
 - [x] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

**Comment by huafengw** [comment]
Test pass: https://travis-ci.org/huafengw/incubator-gearpump/builds/209307018

File: core/src/main/scala/org/apache/gearpump/cluster/master/AppManager.scala [hunk 1]
```diff
@@ -231,35 +231,38 @@ private[cluster] class AppManager(kvService: ActorRef, launcher: AppMasterLaunch
...
-            appResultListeners.getOrElse(appId, List.empty).foreach{ client =>
-              client ! ApplicationFailed(appId, error)
-            }
-          case terminated@ApplicationStatus.TERMINATED =>
-            updatedStatus = appRuntimeInfo.onFinalStatus(timeStamp, terminated)
-          case status =>
-            LOG.error(s"App $appId should not change it's status to $status")
-        }
+        // A dead application should not be able to update status
+        if (!appRuntimeInfo.status.isInstanceOf[ApplicationTerminalStatus]) {
```
Comment (manuzhang) [inline core/src/main/scala/org/apache/gearpump/cluster/master/AppManager.scala]: Do we want to update status if not changed ? Can there be other "false alarm"s ?

**Comment by huafengw** [comment]
I'll refactor status implementation to simple state machine

**Comment by huafengw** [comment]
@manuzhang

**Comment by manuzhang** [comment]
+1, merging

**Pull request closed without merging**